js/src/components/google-ads-account-card/claim-account-button.js

```diff
@@ -41,10 +42,10 @@ const ClaimAccountButton = ( { onClick = noop, ...restProps } ) => {
 
 		onClick( event );
 	};
-
 	return (
 		<AppButton
 			{ ...restProps }
+			loading={ loading || ! inviteLink }
```

Comment:
Using the `inviteLink` value as part of the loading value is problematic, because it is possible for the `inviteLink` to be an empty string, (ie. `inviteLink: ""`). The case of the `inviteLink` being empty seems to be related to a strange race condition.

When no account is connected, the initial state for account-status will be:

```json
{
    "has_access": false,
    "step": "",
    "invite_link": ""
}
```

To create a new account, we call `upsertAdsAccount()`, which internally makes a direct POST request (bypassing actions altogether), followed by refetching the Ads account and the new Ads account status.

This is where a race condition can happen because the store will still have the initial account-status with an `invite_link: ""` and `has_access: false` but the Ads account will be resolved, causing this logic to be true:

```js
const shouldClaimGoogleAdsAccount = Boolean(
	googleAdsAccount?.id && hasAccess === false
);
```

I think the safest thing to do is to either invalidate the account-status data whenever we're creating a new account, or add a condition to `shouldClaimGoogleAdsAccount` that waits for the `upsertAdsAccount()` `loading` value to be `false`. Here's an example:

```diff
diff --git a/js/src/components/google-ads-account-card/claim-account-button.js b/js/src/components/google-ads-account-card/claim-account-button.js
index e9afa0a78..b53ded9c9 100644
--- a/js/src/components/google-ads-account-card/claim-account-button.js
+++ b/js/src/components/google-ads-account-card/claim-account-button.js
@@ -45,7 +45,7 @@ const ClaimAccountButton = ( { onClick = noop, loading, ...restProps } ) => {
 	return (
 		<AppButton
 			{ ...restProps }
-			loading={ loading || ! inviteLink }
+			loading={ loading }
 			eventName="gla_open_ads_account_claim_invitation_button_click"
 			eventProps={ getEventProps() }
 			onClick={ handleClaimAccountClick }
diff --git a/js/src/components/google-combo-account-card/connected-google-combo-account-card.js b/js/src/components/google-combo-account-card/connected-google-combo-account-card.js
index ab2a4eab8..536db2912 100644
--- a/js/src/components/google-combo-account-card/connected-google-combo-account-card.js
+++ b/js/src/components/google-combo-account-card/connected-google-combo-account-card.js
@@ -48,7 +48,7 @@ const ConnectedGoogleComboAccountCard = () => {
 	const { invalidateResolution } = useAppDispatch();
 	const { googleAdsAccount } = useGoogleAdsAccount();
 	const { hasAccess, step } = useGoogleAdsAccountStatus();
-	const [ upsertAdsAccount, { action } ] = useUpsertAdsAccount();
+	const [ upsertAdsAccount, { action, loading } ] = useUpsertAdsAccount();
 
 	const finalizeAdsAccountCreation =
 		hasAccess === true && step === 'conversion_action';
@@ -73,7 +73,7 @@ const ConnectedGoogleComboAccountCard = () => {
 	// @TODO: edit mode implementation in 2605
 	const editMode = false;
 	const shouldClaimGoogleAdsAccount = Boolean(
-		googleAdsAccount?.id && hasAccess === false
+		! loading && googleAdsAccount?.id && hasAccess === false
 	);
 
 	const hasExistingGoogleMCAccounts = existingGoogleMCAccounts?.length > 0;
```

Reply:
Handled in [43c5ab9](https://github.com/woocommerce/google-listings-and-ads/pull/2644/commits/43c5ab9cfc4e29db6454d1462da52c134acf668e) and [6e2b00d](https://github.com/woocommerce/google-listings-and-ads/pull/2644/commits/6e2b00df921ef54087121787981d2c333815a959).